From 8e072f794744bd17c57cceabb3884d3f0f6a1602 Mon Sep 17 00:00:00 2001
From: Thomas Blume <Thomas.Blume@suse.com>
Date: Fri, 16 May 2025 14:27:10 +0200
Subject: [PATCH] dont allow suid and dev set on fs resize

[Salvatore Bonaccorso: Backport to 2.28, which does not contain 2b55f566d422
("fs: Add support for NILFS2 filesystem") from 3.0 which generalized the
bd_fs_mount() call in preparation for other filesystems.]
---
 src/plugins/fs/generic.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/src/plugins/fs/generic.c
+++ b/src/plugins/fs/generic.c
@@ -409,7 +409,7 @@ static gboolean xfs_resize_device (const
                              "before resizing it.", device);
                 return FALSE;
             }
-            ret = bd_fs_mount (device, mountpoint, "xfs", NULL, NULL, error);
+            ret = bd_fs_mount (device, mountpoint, "xfs", "nosuid,nodev", NULL, error);
             if (!ret) {
                 g_prefix_error (error, "Failed to mount '%s' before resizing it: ", device);
                 return FALSE;
